### PR TITLE
Add Support for Overwrite Prompt option in save dialog

### LINKED ISSUE #7030: Add option to disable overwrite prompt with SaveFileDialog
**Is your feature request related to a problem? Please describe.**
I'd like to open a Save File dialog that behaves more as a "Open or Create File" dialog, so that users can either select an existing file, or create a new file.

Right now when you select an existing file, the dialog asks if you want to overwrite the file, which confuses some users and makes them think the file will be emptied.

**Describe the solution you'd like**
I'd like to have a bool property in `SaveFileDialog` that lets me turn off the overwrite prompt.

Windows Forms and WPF has this property under the name [OverwritePrompt](https://docs.microsoft.com/en-us/dotnet/api/microsoft.win32.savefiledialog.overwriteprompt?view=windowsdesktop-6.0). It is `true` by default.

GTK has a function [gtk_file_chooser_set_do_overwrite_confirmation](https://people.gnome.org/~shaunm/girdoc/C/Gtk.FileChooser.set_do_overwrite_confirmation.html). The overwrite prompt is disabled by default.

Because of the differences in default behavior between systems, perhaps a `bool?` would be more appropriate with `null` using the default behavior of the GUI framework.

I don't know if a similar property is available in other GUI frameworks.

### PR DESCRIPTION
**Pull request by emmauss**
## What does the pull request do?
Add OverwritePrompt property to SaveFileDialog. Native Windows, native GTK and managed save dialogs have all been updated to support it.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #7030

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018493-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls/SystemDialog.cs [hunk 1]
```diff
@@ -52,6 +52,11 @@ public class SaveFileDialog : FileDialog
         /// </summary>
         public string? DefaultExtension { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to display a warning if the user specifies the name of a file that already exists.
+        /// </summary>
+        public bool? OverwritePrompt { get; set; }
```
Comment (robloo) [inline src/Avalonia.Controls/SystemDialog.cs:58]: I think the name needs to change:

`IsOverwritePromptShown` or
`ShowOverwritePrompt`

The name `OverwritePrompt` doesn't work for a bool value and implies that you are setting the prompt object itself.

File: src/Avalonia.Dialogs/ManagedFileChooserViewModel.cs [hunk 2]
```diff
@@ -17,6 +17,7 @@ internal class ManagedFileChooserViewModel : InternalViewModelBase
         private readonly ManagedFileDialogOptions _options;
         public event Action CancelRequested;
         public event Action<string[]> CompleteRequested;
+        public event Action<string> ShowOverwritePrompt;
```
Comment (robloo) [inline src/Avalonia.Dialogs/ManagedFileChooserViewModel.cs:20]: This then would simply be `OverwritePrompt`

File: src/Avalonia.Controls/SystemDialog.cs (hunk 1 above)
Comment (emmauss) [reply src/Avalonia.Controls/SystemDialog.cs]: Addressed.

File: src/Avalonia.Dialogs/ManagedFileChooserViewModel.cs (hunk 2 above)
Comment (emmauss) [reply src/Avalonia.Dialogs/ManagedFileChooserViewModel.cs]: Addressed

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018497-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Windows/Avalonia.Win32/SystemDialogImpl.cs [hunk 3]
```diff
@@ -67,6 +67,16 @@ internal class SystemDialogImpl : ISystemDialogImpl
 
                     frm.SetFileTypeIndex(0);
 
+                    if (dialog is SaveFileDialog saveFileDialog)
+                    {
+                        var overwritePrompt = saveFileDialog.ShowOverwritePrompt ?? true;
+
+                        if (!overwritePrompt)
+                        {
+                            frm.SetOptions(frm.Options &~ FILEOPENDIALOGOPTIONS.FOS_OVERWRITEPROMPT);
```
Comment (kekekeks) [inline src/Windows/Avalonia.Win32/SystemDialogImpl.cs:76]: It's better to move where the rest of option calculation happens - https://github.com/AvaloniaUI/Avalonia/blob/afc00162984007c345a6731802fabfdb56842c80/src/Windows/Avalonia.Win32/SystemDialogImpl.cs#L39

Comment (emmauss) [reply src/Windows/Avalonia.Win32/SystemDialogImpl.cs]: Addressed

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018503-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by kekekeks — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018504-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls/SystemDialog.cs (hunk 1 above)
Comment (robloo) [reply src/Avalonia.Controls/SystemDialog.cs]: Note that this deviates from WPF but I think its justified. 

https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.savefiledialog.overwriteprompt?view=windowsdesktop-6.0

File: src/Avalonia.Controls/SystemDialog.cs [hunk 4]
```diff
@@ -52,6 +52,11 @@ public class SaveFileDialog : FileDialog
         /// </summary>
         public string? DefaultExtension { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to display a warning if the user specifies the name of a file that already exists.
+        /// </summary>
+        public bool? ShowOverwritePrompt { get; set; }
```
Comment (robloo) [inline src/Avalonia.Controls/SystemDialog.cs:58]: I'm curious why is this nullable too? Doesn't seem to be required and isn't in WPF

Comment (chylex) [reply src/Avalonia.Controls/SystemDialog.cs:58]: In the original issue, I noted that operating systems (or their respective shells) differ in default behavior if the property is unset. WPF is Windows only, so it doesn't need to care about the default behavior on any other OS.

**Pull request merged**